evalml/tests/data_checks_tests/test_datetime_format_data_check.py

```diff
@@ -25,11 +33,21 @@ def test_datetime_format_data_check_typeerror_uneven_intervals(
     else:
         dates = pd.date_range("2021-01-01", periods=30)
 
-    if uneven:
+    if missing:
         # Skips 2021-01-30 and appends 2021-01-31, skipping a date and triggering the error
         dates = pd.date_range("2021-01-01", periods=29).append(
             pd.date_range("2021-01-31", periods=1)
         )
+    if uneven:
+        dates = pd.DatetimeIndex(
```

Comment:
Do we know what will happen if there is a non-standard frequency?  Like a 3-day frequency or a 30 minute frequency?

Reply:
It seems to work the same way, but I added another test to be sure!